### PR TITLE
refactor IMDB tutorial

### LINKED ISSUE #190: tutorial: IMDB Dataset preprocessing
- [x] jupyter notebook purifying
- [x] .py scripts purifing
- [x] correct typo in tutorials/ImdbDataset/Prepare_IMDB_Data.ipynb: 1 -> rule matches, 0 -> rule doesn't match
- [x] rename ImdbDataset to imbd_dataset to be more consistent
- [x] fix hard coded paths

### PR DESCRIPTION
**Pull request by marina-sp**
closes #190

### CODE REVIEW

**Comment by AndSt** [comment]
I think, to be consistent we should already include the train/dev /test split here. What do you think?

File: tutorials/imdb_dataset_weak_labeling/Prepare_weak_IMDB_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,1774 @@
...
+   "execution_count": 59,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-03-25T21:21:14.291710Z",
+     "start_time": "2021-03-25T21:20:49.165726Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "rule_matches, mapping_rules_labels = transform_rule_class_matrix_to_z_t(applied_lfs)"
```
[the remark this answers is not in the document]
Comment (AndSt) [reply tutorials/imdb_dataset_weak_labeling/Prepare_weak_IMDB_data.ipynb:1492]: Also I would do the same for the Z matrix, to be consistent with Minio :)

**Comment by marina-sp** [comment]
> I think, to be consistent we should already include the train/dev /test split here. What do you think?

I was very surprised to see that the tutorial ends with just the preprocessing and does not do the split, so it is a good idea imo. @agsedova should we add this? 

Also i recall seeing the code for this split somewhere. Would you have an idea where this was?

**Comment by AndSt** [comment]
I found this is in a notebook, could be useful:

`
from sklearn.model_selection import train_test_split

new_data_dir = "/Users/andst/devel/sentence_nf/data/imdb"

rest_df, test_df, rest_rule_matches_sparse_z, test_rule_matches_sparse_z = train_test_split(imdb_dataset_raw, rule_matches_sparse, test_size=5000, random_state=42)
train_df, dev_df, train_rule_matches_sparse_z, dev_rule_matches_sparse_z = train_test_split(rest_df, rest_rule_matches_sparse_z, test_size=5000, random_state=42)

dump(train_df, os.path.join(new_data_dir, "train_df.pbz2"))
dump(dev_df, os.path.join(new_data_dir, "dev_df.pbz2"))
dump(test_df, os.path.join(new_data_dir, "test_df.pbz2"))

dump(train_rule_matches_sparse_z, os.path.join(new_data_dir, "train_rule_matches_sparse_z.pbz2"))
dump(dev_rule_matches_sparse_z, os.path.join(new_data_dir, "dev_rule_matches_sparse_z.pbz2"))
dump(test_rule_matches_sparse_z, os.path.join(new_data_dir, "test_rule_matches_sparse_z.pbz2"))

dump(mapping_rules_labels,  os.path.join(new_data_path, "mapping_rules_labels_t.pbz2"))
`

**Comment by agsedova** [comment]
> > I think, to be consistent we should already include the train/dev /test split here. What do you think?
> 
> I was very surprised to see that the tutorial ends with just the preprocessing and does not do the split, so it is a good idea imo. @agsedova should we add this?

It is definitely a good idea :)

**Pull request merged**